### PR TITLE
feat(react-email): Add hover effects

### PR DESCRIPTION
**Pull request by zenorocha**
https://user-images.githubusercontent.com/398893/213109672-b366392e-cc8b-4ced-88fb-ad3bf5429ec9.mp4

### CODE REVIEW

File: packages/react-email/preview/src/components/code-container.tsx [hunk 1]
```diff
@@ -72,19 +74,34 @@ export const CodeContainer: React.FC<Readonly<CodeContainerProps>> = ({
...
+                  }`}
                   onClick={() => setActiveTab(language)}
+                  onHoverStart={() => setHovered(language)}
+                  onHoverEnd={() => setHovered('')}
+                  key={language}
                 >
+                  {isHovered && (
+                    <motion.span
+                      layoutId="nav"
+                      className="absolute left-0 right-0 top-0 bottom-0 bg-slate-5"
```
Comment (bukinoshita) [inline packages/react-email/preview/src/components/code-container.tsx:94]: Let's use `bg-slate-4` for hovered UI elements
```suggestion
                      className="absolute left-0 right-0 top-0 bottom-0 bg-slate-4"
```

File: packages/react-email/preview/src/components/code-container.tsx [hunk 2]
```diff
@@ -72,19 +74,34 @@ export const CodeContainer: React.FC<Readonly<CodeContainerProps>> = ({
...
-                <button
-                  className={`${activeTab !== language && 'opacity-25'}`}
+        <div className="flex">
+          <AnimateSharedLayout>
+            {markups.map(({ language }) => {
+              const isHovered = hovered === language;
+              return (
+                <motion.button
+                  className={`relative py-[10px] px-4 text-xs transition ease-in-out duration-200 ${
+                    activeTab !== language ? 'text-slate-10' : 'text-slate-12'
```
Comment (bukinoshita) [inline packages/react-email/preview/src/components/code-container.tsx:84]: Text should be `slate-11` or `slate-12`.

```suggestion
                    activeTab !== language ? 'text-slate-11' : 'text-slate-12'
```

File: packages/react-email/preview/package.json [hunk 3]
```diff
@@ -24,6 +24,7 @@
     "@radix-ui/react-tooltip": "1.0.2",
     "@react-email/render": "0.0.6",
     "classnames": "2.3.2",
+    "framer-motion": "^8.4.6",
```
Comment (bukinoshita) [inline packages/react-email/preview/package.json:27]: Need to pin the version here

**Review by bukinoshita — APPROVED** [review]
noice ❤️

**Comment by zenorocha** [comment]
Sidebar hover

https://user-images.githubusercontent.com/398893/213114780-495fcaaa-6fad-4a18-be2e-07abfebc13f7.mp4

**Comment by zenorocha** [comment]
Another one:

https://user-images.githubusercontent.com/398893/213118136-de538cf1-a5c8-417f-8277-9d01d6716a97.mp4

**Comment by bukinoshita** [comment]
Can we vertically align this?

CleanShot 2023-01-25 at 13 16 16@2x

**Pull request merged**